packages/flutter_tools/lib/src/compile.dart

```diff
@@ -620,56 +573,46 @@ class DefaultResidentCompiler implements ResidentCompiler {
 
   @override
   Future<CompilerOutput> recompile(
-    String mainPath,
+    Uri mainUri,
     List<Uri> invalidatedFiles, {
     @required String outputPath,
-    String packagesFilePath,
+    @required PackageConfig packageConfig,
   }) async {
-    assert (outputPath != null);
+    assert(outputPath != null);
     if (!_controller.hasListener) {
       _controller.stream.listen(_handleCompilationRequest);
     }
 
     final Completer<CompilerOutput> completer = Completer<CompilerOutput>();
     _controller.add(
-        _RecompileRequest(completer, mainPath, invalidatedFiles, outputPath, packagesFilePath)
+      _RecompileRequest(completer, mainUri, invalidatedFiles, outputPath, packageConfig)
     );
     return completer.future;
   }
 
   Future<CompilerOutput> _recompile(_RecompileRequest request) async {
     _stdoutHandler.reset();
-
-    // First time recompile is called we actually have to compile the app from
-    // scratch ignoring list of invalidated files.
-    PackageUriMapper packageUriMapper;
-    if (request.packagesFilePath != null || packagesPath != null) {
-      packageUriMapper = PackageUriMapper(
-        request.mainPath,
-        request.packagesFilePath ?? packagesPath,
-        fileSystemScheme,
-        fileSystemRoots,
-      );
-    }
-
     _compileRequestNeedsConfirmation = true;
 
     if (_server == null) {
       return _compile(
-        _mapFilename(request.mainPath, packageUriMapper),
+        request.packageConfig.toPackageUri(request.mainUri)?.toString() ?? request.mainUri.toString(),
         request.outputPath,
-        _mapFilename(request.packagesFilePath ?? packagesPath, /* packageUriMapper= */ null),
       );
     }
-
     final String inputKey = Uuid().generateV4();
-    final String mainUri = request.mainPath != null
-        ? _mapFilename(request.mainPath, packageUriMapper) + ' '
-        : '';
-    _server.stdin.writeln('recompile $mainUri$inputKey');
-    globals.printTrace('<- recompile $mainUri$inputKey');
+    final String mainUri = request.packageConfig.toPackageUri(request.mainUri)?.toString()
+      ?? request.mainUri.toString();
+    _server.stdin.writeln('recompile $mainUri $inputKey');
```

Comment:
Note that failures to write to the `stdin` `IOSink` will manifest as an error completion of the `stdin.done` future. Best practice would be to attach appropriate error handling there. (Maybe just ignore and let the exit code future handle it.)

Reply:
Not sure what you're asking here - should I ignore this for now since its a pre-existing issue?

Reply:
Yeah, just wanted to point it out. It's an issue wherever we spawn a subprocess and write to stdin.